pkg/common/bitmap/bitmap.go

```diff
@@ -47,11 +76,51 @@ func (n *Bitmap) Iterator() Iterator {
 	return &BitmapIterator{i: 0, bm: n}
 }
 
+func wordnum(i uint64) uint64 {
+	return i / bits_per_bitmapword
+}
+
+func bitnum(i uint64) uint64 {
+	return i % bits_per_bitmapword
+}
+
+func rightmost_one_pos_64(word uint64) uint64 {
+	// find out the rightmost_one position. 
+	// Firstly, use eight bits as a group to quickly determine whether there is a 1 in it.
+	// if not, then rightmost_one exists in next group, add up the distance with result and shift the word
+	// if rightmost_one exists in this group, get the distance directly from a pre-made hash table
+	var result uint64
+	for {
+		if (word & 255) == 0 {
+			word >>= 8
+			result += 8
+		} else {
+			break
+		}
+	}
+	result += uint64(rightmost_one_pos_8[word&255])
+	return result
+}
+
 func (itr *BitmapIterator) HasNext() bool {
-	for ; itr.i < uint64(itr.bm.len); itr.i++ {
-		if itr.bm.Contains(itr.i) {
+	// if the uint64 is 0, move forward to next word
+	// if the uint64 is not 0, then calculate the rightest_one position in a word and add it to itr.i
+	// either case loop over words not bits.
+	nwords := (itr.bm.len-1)/64 + 1
+	current_word := wordnum(itr.i)
+	mask := ((bitmask)(0) | 0xFFFFFFFF) << bitnum(itr.i) // ignore bits before check before
```

Comment:
Are you using uint32 or uint64?    I think this 0xFFF... does not have enough F in it.

Reply:
A better implementation has been used